jersey/connector/src/main/java/io/helidon/jersey/connector/HelidonConnector.java

```diff
@@ -78,42 +77,60 @@ class HelidonConnector implements Connector {
     private final Proxy proxy;
 
     @SuppressWarnings("unchecked")
-    HelidonConnector(Client client, Configuration config) {
+    HelidonConnector(Client client, Configuration configuration) {
         // create underlying HTTP client
-        Map<String, Object> properties = config.getProperties();
+        Map<String, Object> properties = configuration.getProperties();
         WebClientConfig.Builder builder = WebClientConfig.builder();
 
-        // use config for client
-        Config helidonConfig = helidonConfig(config).orElse(Config.empty());
-        builder.config(helidonConfig);
+        // use config from property first then registry
+        var helidonConfig = configuration.getProperty(HelidonProperties.CONFIG);
+        if (helidonConfig != null) {
+            if (helidonConfig instanceof Config) {
+                builder.config((Config) helidonConfig);
+            } else {
+                LOGGER.log(System.Logger.Level.WARNING,
+                           String.format("Ignoring Helidon Connector config at '%s'", HelidonProperties.CONFIG));
+                builder.config(configFromRegistry());
+            }
+        } else {
+            builder.config(configFromRegistry());
+        }
 
         // proxy support
-        proxy = ProxyBuilder.createProxy(config).orElse(Proxy.create());
+        proxy = ProxyBuilder.createProxy(configuration).orElse(Proxy.create());
 
-        // possibly override config with properties
+        // possibly override config with properties, if properly defined
         if (properties.containsKey(CONNECT_TIMEOUT)) {
-            builder.connectTimeout(Duration.ofMillis(getValue(properties, CONNECT_TIMEOUT, DEFAULT_TIMEOUT)));
+            Integer connectTimeout = getValue(properties, CONNECT_TIMEOUT, Integer.class);
+            if (connectTimeout != null) {
+                builder.connectTimeout(Duration.ofMillis(connectTimeout));
+            }
         }
         if (properties.containsKey(READ_TIMEOUT)) {
-            builder.readTimeout(Duration.ofMillis(getValue(properties, READ_TIMEOUT, DEFAULT_TIMEOUT)));
+            Integer readTimeout = getValue(properties, READ_TIMEOUT, Integer.class);
+            if (readTimeout != null) {
```

Comment:
(If you're going to always call `getValue` and check to see whether it worked by checking for `null`, then you can eliminate the `containsKey` call.)

Reply:
If it's there it can still return `null` if `getValue` is unable to convert the value to the type requested. In fact, it should at least log that condition and it isn't yet.

Reply:
OK!